### PR TITLE
added collection call options for 3.6 compatibility

### PR DESCRIPTION
**Pull request by gpernot**
Just add bypass_document_validation= and session= options to collection calls for 3.6 compatibility

### CODE REVIEW

File: mongomock/collection.py [hunk 1]
```diff
@@ -371,18 +371,21 @@ def _remove_key(self, doc, key):
             sub_doc = sub_doc[part]
         del sub_doc[key_parts[-1]]
 
-    def update_one(self, filter, update, upsert=False):
+    def update_one(self, filter, update, upsert=False, bypass_document_validation=False,
```
Comment (vmalloc) [inline mongomock/collection.py]: what's array_filters? does it have noticeable side effects when used?

**Review by vmalloc — CHANGES_REQUESTED** [review]

**Comment by gpernot** [comment]
Hi,

> 
> - def update_one(self, filter, update, upsert=False):
> + def update_one(self, filter, update, upsert=False, bypass_document_validation=False,what's
> array_filters? does it have noticeable side effects when used?

actually, it does have side effects : it updates array elements matching the filters.
https://docs.mongodb.com/manual/reference/operator/update/positional-filtered/#up._S_[%3Cidentifier%3E]

i have no clue on how to implement this is mongomock, maybe we should just drop the array_filters parameter for now ?

**Comment by vmalloc** [comment]
@gpernot yes - if you can't provide a matching implementation it's better to not support it at all, as per the guidelines

**Review by gpernot — COMMENTED** [review]
I only left the 'session' parameter in latest patch. I guess the implementation is fine, as mongomock is basically synchronous.

**Review by vmalloc — APPROVED** [review]

**Comment by vmalloc** [comment]
@gpernot I can't merge this since the Travis build is failing. Can you take a look?

**Comment by gpernot** [comment]
11 juin 2018 07:54 "Rotem Yaari" <notifications@github.com> a écrit:
> @gpernot I can't merge this since the Travis build is failing. Can you take a look?

travis fails on :
./mongomock/__init__.py:79:1: E402 module level import not at top of file

and i haven't modified __init__.py at all

maybe you have more experience than me on this ?

**Comment by vmalloc** [comment]
Ok, will take a look. Likely I'll have to ask you to rebase on develop in case I need to fix anything.

**Comment by vmalloc** [comment]
@gpernot I fixed the flake8 settings on develop. Can you please rebase?

**Comment by vmalloc** [comment]
Thanks!

**Pull request merged**